api/app/src/routes/util.ts

```diff
@@ -40,12 +44,20 @@ export const getUserIdFromHeaders = (req: Request): string | undefined =>
   req.header("userId") as string | undefined;
 export const getUserId = (req: Request): string | undefined =>
   req.query.userId as string | undefined;
-export const getUserIdOrFail = (req: Request): string => {
+export const getUserIdFromQueryOrFail = (req: Request): string => {
   const userId = getUserId(req);
   if (!userId) throw new BadRequestError("Missing userId query param");
   return userId as string;
 };
 
+export const getProviderMapFromConnectUserOrFail = (connectedUser: ConnectedUser, provider: ProviderOptions) => {
+  if (!connectedUser.providerMap) throw new UnauthorizedError();
+  const providerData = connectedUser.providerMap[provider];
+  if (!providerData) throw new UnauthorizedError();
+
+  return providerData;
+}
```

Comment:
Good job identifying and reusing code! 🙌 

I just suggest we keep these util functions to a minimal - usually not domain-related, otherwise we can quickly have a ton of them; as well, exposing the ConnectedUser's internals adds coupling across the app.

In this case, since we need this behavior across multiple files, maybe move this function to be a command related to ConnectedUser? It could throw `NotFoundError` and we have the route rethrowing it as an `UnauthorizedError`.

Also, the function returns a Provider's data, not the map, so maybe rename to `getProviderDataOrFail()` or something similar?